pkg/minikube/download/download.go

```diff
@@ -91,3 +99,32 @@ func withinUnitTest() bool {
 
 	return flag.Lookup("test.v") != nil || strings.HasSuffix(os.Args[0], "test")
 }
+
+// lockDownload locks `file` if possible and returns a releaser that must be called to release the lock.
+func lockDownload(file string) (mutex.Releaser, error) {
+	type retPair struct {
+		mutex.Releaser
+		error
+	}
+	lockChannel := make(chan retPair)
+
+	go func() {
```

Comment:
Are we starting a separate goroutine to acquire the mutex only to be able to display `"Another minikube instance is downloading dependencies... "` message if the mutex is already locked? We still block forever on line 128.  May be it makes sense to simplify the code and get rid of extra goroutine, channel and select?

Reply:
got it. a comment about this internal mutex timeout would be great here, actually.  Regarding the message, may be just a couple of logs before/immediately after getting the mutex? something like `acquiring the fs lock..`/`acquired`?

Reply:
lets keep the message for the user so they don't wait without any explaintion, our users like to be updated when something is waiting longer than a few sceonds (exmaple the complain about kubeadm init)

Reply:
for Func Names not related to this PR I would suggest making follow up PRs to keep this PR as small as possible to review